feature_engine/base_transformers.py

```diff
@@ -97,6 +97,9 @@ def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None) -> pd.DataFrame:
         # find or check for numerical variables
         self.variables_ = _find_or_check_numerical_variables(X, self.variables)
 
+        # save input features
+        self.input_features_: List[Union[str, int]] = X.columns.tolist()
```

Comment:
For some transformers, it would not be only the input features, but the input features plus the features that are added or removed. So I would use a more neutral name here to unify across classes like feature_names_